x-pack/legacy/plugins/siem/public/components/charts/barchart.tsx

```diff
@@ -102,19 +116,54 @@ BarChartBase.displayName = 'BarChartBase';
 interface BarChartComponentProps {
   barChart: ChartSeriesData[] | null | undefined;
   configs?: ChartSeriesConfigs | undefined;
+  stackByField?: string;
 }
 
-export const BarChartComponent: React.FC<BarChartComponentProps> = ({ barChart, configs }) => {
+const NO_LEGEND_DATA: LegendItem[] = [];
+
+export const BarChartComponent: React.FC<BarChartComponentProps> = ({
+  barChart,
+  configs,
+  stackByField,
+}) => {
   const { ref: measureRef, width, height } = useThrottledResizeObserver();
+  const legendItems: LegendItem[] = useMemo(
+    () =>
+      barChart != null && stackByField != null
+        ? barChart.map((d, i) => ({
+            color: d.color ?? i < defaultLegendColors.length ? defaultLegendColors[i] : undefined,
+            dataProviderId: escapeDataProviderId(
+              `draggable-legend-item-${uuid.v4()}-${stackByField}-${d.key}`
+            ),
+            field: stackByField,
+            value: d.key,
+          }))
+        : NO_LEGEND_DATA,
+    [barChart, stackByField]
+  );
+
   const customHeight = get('customHeight', configs);
   const customWidth = get('customWidth', configs);
   const chartHeight = getChartHeight(customHeight, height);
   const chartWidth = getChartWidth(customWidth, width);
 
   return checkIfAnyValidSeriesExist(barChart) ? (
-    <WrappedByAutoSizer ref={measureRef} height={chartHeight}>
-      <BarChartBase height={chartHeight} width={chartHeight} data={barChart} configs={configs} />
-    </WrappedByAutoSizer>
+    <EuiFlexGroup gutterSize="none">
+      <EuiFlexItem grow={true}>
+        <WrappedByAutoSizer ref={measureRef} height={chartHeight}>
+          <BarChartBase
+            configs={configs}
+            data={barChart}
+            forceHiddenLegend={stackByField != null}
+            height={chartHeight}
+            width={chartHeight}
+          />
+        </WrappedByAutoSizer>
+      </EuiFlexItem>
+      <LegendFlexItem grow={false}>
+        {stackByField != null && <DraggableLegend legendItems={legendItems} height={height} />}
```

Comment:
I do not think that we need the condition `stackByField != null` because we already manage it in the useMemo and inside of DraggableLegend